javascripts/2048.js

```diff
@@ -1,32 +1,281 @@
 $(document).ready(function() {
   console.log('ready!');
+
+  placeFirstTiles();
+
   $('body').keydown(function(event){
     var arrow_keys = [37, 38, 39, 40];
-    if(arrow_keys.indexOf(event.which) > -1) {
-      var tile = $('.tile');
-      moveTile(tile, event.which);
-      event.preventDefault();
+
+    if(arrow_keys.indexOf(event.which) > -1) { // meaning the key that was pressed is in the array
+      var tile = $('.tile');                   // all of the tiles
+      moveTile(tile, event.which);             // move all of the tiles in the arrow direction
+      event.preventDefault();                  // prevent default arrow key browser movement
     }
-  })
-})
+  });
+});
+
+// global variables
+var score = 0;
+var ALL_GRID_POSITIONS = ["r0,c0", "r0,c1", "r0,c2", "r0,c3", "r1,c0", "r1,c1", "r1,c2", "r1,c3", "r2,c0", "r2,c1", "r2,c2", "r2,c3", "r3,c0", "r3,c1", "r3,c2", "r3,c3"];
+
+// cloning grid positions necessary to prevent mutating original varibale
+function cloneGridPositions() {
+  var clone = ALL_GRID_POSITIONS.slice(0);
+  return clone;
+}
+
+function placeFirstTiles() {
+  var grid_options = cloneGridPositions();
+
+  // place 2 tiles to start game
+  for (i = 0; i < 2; i++) {
+    // pick random coordinate
+    var random_index = getRandomIntInclusive(0, grid_options.length - 1);
+    var new_tile_position = grid_options[random_index]; // "r0,c0"
+
+    addTileFromCoordinate(new_tile_position, 2);
+
+    // remove from array
+    grid_options.remove(new_tile_position);
+  }
+}
+
+function addTileFromCoordinate(coordinate, data_val) {
+  var new_row = coordinate.split(",")[0]; // "r0"
+  var new_col = coordinate.split(",")[1]; // "c0"
+  var new_tile = $("<div>");
+  new_tile.addClass("tile");
+  new_tile.attr({
+        "data-row" : new_row,
+        "data-col" : new_col,
+        "data-val" : data_val
+    });
+  new_tile.html(data_val);
+  // add tile to board
+  $("#gameboard").append(new_tile);
+  new_tile.attr('combined','false');
+  pop(new_tile);
+}
+
+function findTile(row_position, col_position) {
+  return $('.tile[data-row="r' + row_position + '"][data-col="c' + col_position + '"]');
+}
+
+function getPositionNum(active_tile, axis) {
+  return parseInt(active_tile.getAttribute(axis)[1]);
+}
+
+function checkNextSpace(active_tile, direction) {
+  var data_row_num  = getPositionNum(active_tile, 'data-row');
+  var data_col_num  = getPositionNum(active_tile, 'data-col');
+  var data_val      = parseInt(active_tile.getAttribute('data-val'));
+
+  switch(direction) {
+    case 38: var next_row_num = data_row_num - 1 ; break; // up
```

Comment:
Instead of commenting these numeric values, I would use a constant. Then you don't need the comment and it works even when you reuse them in the next section.